mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,47 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last and 'last_ckpt' in self.runtime_info:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            self._publish_model(runner, last_ckpt)
+
+        if self.save_best is not None:
+            if len(self.key_indicators) == 1:
+                if self.best_ckpt_path is not None:
+                    self._publish_model(runner, self.best_ckpt_path)
+            else:
+                for key, best_ckpt in self.best_ckpt_path_dict.items():
+                    self._publish_model(runner, best_ckpt)
+
+    def _publish_model(self, runner, out_file: str) -> None:
+
+        checkpoint = _load_checkpoint(out_file)
+        removed_keys = []
+        for key in checkpoint.keys():
+            if key not in self.published_keys:
+                removed_keys.append(key)
+                checkpoint.pop(k)
+        if removed_keys:     
+            print_log(
+                f'Key {removed_keys} will be removed because they are not '
+                'found in published_keys. If you want to keep them, '
+                f'please set `{removed_keys}` in published_keys',
+                logger='current')
+        print(runner.timestamp)
+        final_file = osp.splitext(out_file)[0] + f'-{runner.timestamp}.pth'
+        torch.save(checkpoint, final_file)
```

Comment:
It could be better to use save_checkpoint(`from mmengine.runner import save_checkpoint`) to save the published model to make sure the checkpoint can be uploaded to non-local backend.